src/sage/rings/function_field/drinfeld_modules/finite_drinfeld_module.py

```diff
@@ -303,31 +503,27 @@ def frobenius_norm(self):
             The Frobenius norm is computed using the formula, by
             Gekeler, given in [MS2019]_, Section 4, Proposition 3.
         """
-        self._check_rank_two()
-        L = self._base.over(self._Fq)
-        # Notations from Schost-Musleh:
-        if self._frobenius_norm is None:
-            n = L.degree_over(self._Fq)
-            d = self.characteristic().degree()
-            m = n // d
-            delta = self._gen[2]
-            norm = L(delta).norm()
-            char = self.characteristic()
-            self._frobenius_norm = ((-1)**n) * (char**m) / norm
+        if self._frobenius_norm is not None:
+            return self._frobenius_norm
+        K = self.base_over_constants_field()
+        Fq = self._Fq
+        Kq = K.over(Fq)
+        n = K.degree(Fq)
+        char = self.characteristic()
+        self._frobenius_norm = ((-1)**n)*(char**(n/char.degree())) \
+                               / Kq(self.coefficients()[-1]).norm()
```

Comment:
I believe casting `Kq = K.over(Fq)` is useless (I'm really sorry for missing that earlier!), as, by definition, `K` is already an extension of `Fq`. I also reformatted a bit:

```suggestion
        K = self.base_over_constants_field()
        Fq = self._Fq
        n = K.degree(Fq)
        char = self.characteristic()
        norm = K(self.coefficients()[-1]).norm()
        self._frobenius_norm = ((-1)**n)*(char**(n/char.degree())) / norm
        return self._frobenius_norm
```

Reply:
In testing, it seems to give the wrong norm, so the casting seems necessary (and seems to fix it). But perhaps there is another issue I'm missing that should be corrected.

Reply:
Same number of lines; more readable. Can you do the following change?

```
        norm = K(self.coefficients()[-1]).norm()
        self._frobenius_norm = ((-1)**n) * (char**(n/char.degree())) / norm
```

Reply:
Thank you @ymusleh :heart: